light-zk.js/src/utxo.ts

```diff
@@ -791,6 +821,61 @@ export class Utxo {
     }
   }
 
+  // TODO: unify compressed and includeAppData into a parsingConfig or just keep one
+  /**
+   * @description Decrypts an utxo from an array of bytes by checking the UTXO prefix hash, the last 24 bytes are the nonce.
+   * @param {any} poseidon
+   * @param {Uint8Array} encBytes
+   * @param {Account} account
+   * @param {number} index
+   * @returns {Utxo | boolean }
+   */
+  static async decrypt({
+    poseidon,
+    encBytes,
+    account,
+    index,
+    merkleTreePdaPublicKey,
+    aes,
+    commitment,
+    appDataIdl,
+    compressed = true,
+    assetLookupTable,
+    verifierProgramLookupTable,
+  }: {
+    poseidon: any;
+    encBytes: Uint8Array;
+    account: Account;
+    index: number;
+    merkleTreePdaPublicKey?: PublicKey;
+    aes: boolean;
+    commitment: Uint8Array;
+    appDataIdl?: Idl;
+    compressed?: boolean;
+    assetLookupTable: string[];
+    verifierProgramLookupTable: string[];
+  }): Promise<Utxo | boolean> {
+    const prefixBytes = encBytes.slice(0, PREFIX_LENGTH);
+    if (aes && this.checkPrefixHash({ account, commitment, prefixBytes })) {
+      const utxo = await Utxo.decryptUnchecked({
+        poseidon,
+        encBytes,
+        account,
+        index,
+        merkleTreePdaPublicKey,
+        aes,
+        commitment,
+        appDataIdl,
+        compressed,
+        assetLookupTable,
+        verifierProgramLookupTable,
+      });
+      if (!utxo)
+        return true; // prefixHash matches but decryption fails so utxo is null -> Returns TRUE (COLLISION)
+      else return utxo; // prefixHash matches and decryption succeeds so utxo is good -> Returns UTXO (VALID)
+    } else return false; // prefixHash doesn't match -> Returns FALSE (NO COLLISION)
```

Comment:
I don't like it either. 
We can revert it back to null or we can convert resulting expression `(Utxo | boolean)` to something more strongly typed like Result<Utxo, UtxoDecryptionError>. 
Matteo used `bool` result  in his collision test, but I'm not sure why this information is important for us, @ananas-block can you comment on this please? https://github.com/Lightprotocol/light-protocol/blob/matteo/utxo-decryption-improvements/light-zk.js/tests/utxo.test.ts#L728
If it's really needed, I'll go with the Result<Utxo, UtxoDecryptionError> refactoring, otherwise I'll just return `null`.

Reply:
- hm, I don't remember the specific reason why we differentiated between the errors and did not just return null.
- In general, I prefer explicit error handling, the more explicit the better.
- In my opinion returning enum instead of a boolean is a good improvement.

Reply:
The collision counter is part of the benchmarking.
Every prefix collision will slow down the process because it means that we tried to decrypt a utxo and werent successful.
https://github.com/Lightprotocol/light-protocol/blob/matteo/utxo-decryption-improvements/light-zk.js/tests/utxo.test.ts#L728